sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/NotificationHookLiveTests.cs

```diff
@@ -0,0 +1,389 @@
+
+            await using var disposableHook = await DisposableNotificationHook.CreateHookAsync(adminClient, hookToCreate);
+
+            var options = new GetHooksOptions();
+
+            if (populateOptionalMembers)
+            {
+                options.HookNameFilter = hookNameFilter;
+            }
+
+            var hookCount = 0;
+
+            await foreach (NotificationHook hook in adminClient.GetHooksAsync(options))
+            {
+                Assert.That(hook.Id, Is.Not.Null.And.Not.Empty);
+                Assert.That(hook.Name, Is.Not.Null.And.Not.Empty);
+                Assert.That(hook.Administrators, Is.Not.Null.And.Not.Empty);
+                Assert.That(hook.Administrators.Any(admin => admin == null || admin == string.Empty), Is.False);
+                Assert.That(hook.Description, Is.Not.Null);
+                Assert.That(hook.ExternalLink, Is.Not.Null);
+
+                if (populateOptionalMembers)
+                {
+                    Assert.That(hook.Name.Contains(hookNameFilter));
+                }
+
+                if (hook is EmailNotificationHook)
+                {
+                    var emailHook = hook as EmailNotificationHook;
+
+                    Assert.That(emailHook.EmailsToAlert, Is.Not.Null);
+                }
+                else
+                {
+                    var webHook = hook as WebNotificationHook;
+
+                    Assert.That(webHook, Is.Not.Null);
+                    Assert.That(webHook.CertificateKey, Is.Not.Null);
+                    Assert.That(webHook.CertificatePassword, Is.Not.Null);
+                    Assert.That(webHook.Username, Is.Not.Null);
+                    Assert.That(webHook.Password, Is.Not.Null);
+                    Assert.That(webHook.Headers, Is.Not.Null);
+                    Assert.That(webHook.Headers.Values.Any(value => value == null), Is.False);
+                }
+
+                if (++hookCount >= MaximumSamplesCount)
+                {
+                    break;
+                }
+            }
+
+            Assert.That(hookCount, Is.GreaterThan(0));
+        }
+
+        [RecordedTest]
+        public async Task DeleteNotificationHook()
+        {
+            MetricsAdvisorAdministrationClient adminClient = GetMetricsAdvisorAdministrationClient();
+
+            string hookName = Recording.GenerateAlphaNumericId("hook");
+            var hookToCreate = new WebNotificationHook(hookName, "http://contoso.com");
+
+            string hookId = null;
+
+            try
+            {
+                hookId = await adminClient.CreateHookAsync(hookToCreate);
+
+                Assert.That(hookId, Is.Not.Null.And.Not.Empty);
+            }
+            finally
+            {
+                if (hookId != null)
+                {
+                    await adminClient.DeleteHookAsync(hookId);
+
+                    Assert.That(async () => await adminClient.GetHookAsync(hookId), Throws.InstanceOf<RequestFailedException>());
```

Comment:
is there a way to validate that the reason why it failed is because the hook doesn't exist anymore? and not a random request error?

Reply:
We can check the error message. Updated code.